cmd/juju/caas/credential.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2019 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package caas
+
+import (
+	"encoding/hex"
+	"math/rand"
+
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/caas/kubernetes/clientconfig"
+	"github.com/juju/juju/caas/kubernetes/provider"
+	jujucloud "github.com/juju/juju/cloud"
+	"github.com/juju/juju/environs"
+)
+
+const rbacLabelKeyName = provider.RBACLabelKeyName
+
+func ensureCredentialUID(
+	credentialName, credentialUID string,
+	credential jujucloud.Credential,
+) (cred jujucloud.Credential, _ error) {
+
+	newAttr := credential.Attributes()
+	if newAttr == nil {
+		return cred, errors.NotValidf("empty credential %q", credentialName)
+	}
+	newAttr[rbacLabelKeyName] = credentialUID
+	return jujucloud.NewNamedCredential(
+		credentialName, credential.AuthType(), newAttr, credential.Revoked,
+	), nil
+}
+
+type credentialGetter interface {
+	// CredentialForCloud gets credentials for the named cloud.
+	CredentialForCloud(string) (*jujucloud.CloudCredential, error)
+}
+
+func getExistingCredential(store credentialGetter, cloudName, credentialName string) (credential jujucloud.Credential, err error) {
```

Comment:
This is a local credential on the client?
Can we call this getExistingLocalCredential?

Reply:
yes, sure, changed.